components/zcash_protocol/src/lib.rs

```diff
@@ -25,7 +25,8 @@ pub mod memo;
 pub mod value;
 
 /// A Zcash shielded transfer protocol.
-#[derive(Debug, Copy, Clone, PartialEq, Eq, PartialOrd, Ord)]
+#[derive(Debug, Copy, Clone, PartialEq, Eq, PartialOrd, Ord, Hash)]
+#[cfg_attr(feature = "serde", derive(serde::Serialize, serde::Deserialize))]
```

Comment:
What is the stability requirement for the serialized values here? If new enum variants are added, does the order matter in terms of being able to deserialize existing persistent data? Please document such requirements; if there's a way to assign stable identifiers for the serialization, that would be preferable instead of relying upon the ones that are implicitly assigned.